File: test09/distinct_lines.py
#!/usr/bin/python

import sys

num=int(sys.argv[1])
list=[]
n=0
for line in sys.stdin:
    n+=1
    li=line.split(' ')
    
    l=''
    for i in li:
        if i != '' and i !='\n':
            i=i.lower()
            l=l+i+' '
    if l not in list:
        list.append(l)
        
        count=len(list)
        if count==num:
            print('%d distinct lines seen after %d lines read.' % (num, n))
            exit()
print('End of input reached after %d lines read -  %d different lines not seen.' % (n,num))
    

File: test07/distinct_lines.pl
#!/usr/bin/perl

@shuzu=();
$zhi=$ARGV[0];
$n=0;
while ($str=<STDIN>){
$n+=1;
@shu=split " ",$str;
@shushu=();
foreach $ss(@shu){
if ($ss){
push(@shushu,lc($ss));
}
}
$sss=join(' ',@shushu);
if ($sss~~@shuzu){
}
else{
push(@shuzu,$sss);
$size=@shuzu;
if ($size==$zhi){
print "$size distinct lines seen after $n lines read.\n";
exit;
}
}
}
print "End of input reached after $n lines read - $zhi different lines not seen.\n";
